src/Tests/Microsoft.Diagnostics.Monitoring.ExecuteActionApp/Program.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.ExecuteActionApp
+{
+    internal class Program
+    {
+        public static int Main(string[] args)
+        {
+            string testType = args[0];
+
+            string[] additionalArgs = args.Skip(1).ToArray();
+
+            switch (testType)
+            {
+                case "ZeroExitCode":
+                    Assert.Equal(1, args.Length);
+                    return 0;
+
+                case "NonzeroExitCode":
+                    Assert.Equal(1, args.Length);
+                    return -1;
+
+                case "Sleep":
+                    Assert.Equal(2, args.Length);
+                    string delayArg = additionalArgs[0];
+                    int delay = int.Parse(delayArg) + 1000; // Add a second delay to the token cancellation time
```

Comment:
Make the app are basic as possible: don't make the app have assumptions about how it is tested. Just take the sleep time as-is and sleep on it. Have the test itself add whatever padding it needs.